src/Forms/SyncPointsSync.cs

```diff
@@ -466,5 +476,116 @@ private void ButtonFindTextOtherClick(object sender, EventArgs e)
+        {
+            /*It should be possible to change these variables in a autosync settings menu*/
+            int TIME_MARGIN = 5;
+            int MINIMUM_WORDS_IN_PARAGRAPH = 1;
+            int TARGET_PERCENTAGE = 60;
+            labelAutoSyncing.Visible = true;
+            progressAutoSync.Visible = true;
+            AutoSync(_subtitle, _otherSubtitle, TIME_MARGIN, MINIMUM_WORDS_IN_PARAGRAPH, TARGET_PERCENTAGE, (comboBoxSubToSyncLanguage.SelectedItem as ComboBoxItem).Value, (comboBoxOtherSubLanguage.SelectedItem as ComboBoxItem).Value);
+            labelAutoSyncing.Visible = false;
+            progressAutoSync.Visible = false;
+        }
+
+        /*
+            This function will try to sync subtitles automatically by comparing the one subtitle with another google translated one.
+        */
+        private void AutoSync(Subtitle subtitleToSync, Subtitle refrence, int timeMargin, int minimumWordsInParagraph, int targetPercentage, string languageSubtileToSync, string languageRefrence)
+        {
+            //Tasks are used beceause translation process takes a long time
+            Task<Subtitle> taskTranslateOther = Task<Subtitle>.Factory.StartNew(() => TranslateSub(refrence, languageRefrence + "|" + languageSubtileToSync));
+            Task<Subtitle> taskTranslateSubtileToSync = Task<Subtitle>.Factory.StartNew(() => TranslateSub(subtitleToSync, languageSubtileToSync + "|" + languageRefrence));
+
+            int numberOfEqualWords;
+            int numberOfWrongWords;
+            int nextSyncPoint = 0;
+            string[] subtitleWords;
+            string[] translatedSubs;
+
+            Dictionary<int, int> syncPoints = new Dictionary<int, int>();
+
+            Subtitle subtitleWrong = new Subtitle(subtitleToSync, true);
+
+            var otherTranslated = taskTranslateOther.Result;
+
+            for (int z = 0; z < 2; z++)
+            {
+                //for each paragraph subtitle to sync
+                for (int i = 0; i < subtitleWrong.Paragraphs.Count; i++)
+                {
+                    subtitleWords = Regex.Matches(subtitleWrong.Paragraphs[i].Text.ToLower(), @"\w+").Cast<Match>().Select(x => x.Value).ToArray(); //_subtitle.Paragraphs[i].Text.Split(' ');
+                    //for each translated paragraph in range
+                    for (int j = nextSyncPoint; j < otherTranslated.Paragraphs.Count && (subtitleWrong.Paragraphs[i].StartTime.Minutes < otherTranslated.Paragraphs[j].StartTime.Minutes + timeMargin && subtitleWrong.Paragraphs[i].StartTime.Minutes > otherTranslated.Paragraphs[j].StartTime.Minutes - timeMargin); j++)
+                    {
+                        numberOfEqualWords = 0;
+                        numberOfWrongWords = 0;
+                        translatedSubs = Regex.Matches(otherTranslated.Paragraphs[j].Text.ToLower(), @"\w+").Cast<Match>().Select(x => x.Value).ToArray(); //otherTranslated.Paragraphs[j].Text.Split(' ');

```

Comment:
I guess I need help implementing the code into the program. And other points below.

- Preventing the UI from freezing when the code is running.

- Adding a progress bar that is updated properly

- Making the code faster

- Maybe add a options menu for the auto translation